quantized elu: eager mode static handling

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #40284 graph mode: add hardswish inplace handling
* #40111 quantized elu: graph mode handling
* #40104 quantized elu: eager mode QAT handling
* **#40103 quantized elu: eager mode static handling**
* #40100 quantized elu: require observation

Summary:

Add eager mode static quantization handling for quantized ELU.

Test Plan:

```
python test/test_quantization.py TestStaticQuantizedModule.test_elu
python test/test_quantization.py TestPostTrainingStatic.test_activations
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22075081](https://our.internmc.facebook.com/intern/diff/D22075081)

vkuzo committed:
quantized elu: eager mode static handling

Summary:

Add eager mode static quantization handling for quantized ELU.

Test Plan:

```
python test/test_quantization.py TestStaticQuantizedModule.test_elu
python test/test_quantization.py TestPostTrainingStatic.test_activations
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo committed:
Update on "quantized elu: eager mode static handling"


Summary:

Add eager mode static quantization handling for quantized ELU.

Test Plan:

```
python test/test_quantization.py TestStaticQuantizedModule.test_elu
python test/test_quantization.py TestPostTrainingStatic.test_activations
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo committed:
Update on "quantized elu: eager mode static handling"


Summary:

Add eager mode static quantization handling for quantized ELU.

Test Plan:

```
python test/test_quantization.py TestStaticQuantizedModule.test_elu
python test/test_quantization.py TestPostTrainingStatic.test_activations
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22075081](https://our.internmc.facebook.com/intern/diff/D22075081)

[ghstack-poisoned]

jerryzh168 commented on torch/testing/_internal/common_quantization.py:
```diff
@@ -418,10 +418,12 @@ def __init__(self):
     def forward(self, x):
         x = self.quant(x)
         x = self.hardswish(x)
+        x = self.elu(x)
```
there is no dequantize in the end?

vkuzo committed:
Update on "quantized elu: eager mode static handling"


Summary:

Add eager mode static quantization handling for quantized ELU.

Test Plan:

```
python test/test_quantization.py TestStaticQuantizedModule.test_elu
python test/test_quantization.py TestPostTrainingStatic.test_activations
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22075081](https://our.internmc.facebook.com/intern/diff/D22075081)

[ghstack-poisoned]

jerryzh168 commented on torch/nn/quantized/modules/activation.py:
```diff
@@ -123,3 +123,23 @@ def _get_name(self):
         return Hardswish(float(scale), int(zero_point))
+
+class ELU(torch.nn.ELU):
+    r"""This is the quantized equivalent of `torch.nn.ELU`.
```
you can do ":class:`torch.nn.ELU`" to link to the doc I think

vkuzo committed:
Update on "quantized elu: eager mode static handling"


Summary:

Add eager mode static quantization handling for quantized ELU.

Test Plan:

```
python test/test_quantization.py TestStaticQuantizedModule.test_elu
python test/test_quantization.py TestPostTrainingStatic.test_activations
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22075081](https://our.internmc.facebook.com/intern/diff/D22075081)

[ghstack-poisoned]

vkuzo committed:
Update on "quantized elu: eager mode static handling"


Summary:

Add eager mode static quantization handling for quantized ELU.

Test Plan:

```
python test/test_quantization.py TestStaticQuantizedModule.test_elu
python test/test_quantization.py TestPostTrainingStatic.test_activations
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22075081](https://our.internmc.facebook.com/intern/diff/D22075081)

[ghstack-poisoned]

vkuzo committed:
Update on "quantized elu: eager mode static handling"


Summary:

Add eager mode static quantization handling for quantized ELU.

Test Plan:

```
python test/test_quantization.py TestStaticQuantizedModule.test_elu
python test/test_quantization.py TestPostTrainingStatic.test_activations
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22075081](https://our.internmc.facebook.com/intern/diff/D22075081)

[ghstack-poisoned]

vkuzo committed:
Update on "quantized elu: eager mode static handling"


Summary:

Add eager mode static quantization handling for quantized ELU.

Test Plan:

```
python test/test_quantization.py TestStaticQuantizedModule.test_elu
python test/test_quantization.py TestPostTrainingStatic.test_activations
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22075081](https://our.internmc.facebook.com/intern/diff/D22075081)

[ghstack-poisoned]

vkuzo committed:
Update on "quantized elu: eager mode static handling"


Summary:

Add eager mode static quantization handling for quantized ELU.

Test Plan:

```
python test/test_quantization.py TestStaticQuantizedModule.test_elu
python test/test_quantization.py TestPostTrainingStatic.test_activations
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22075081](https://our.internmc.facebook.com/intern/diff/D22075081)

[ghstack-poisoned]
